gateway/src/shard/processor/socket_forwarder.rs

```diff
@@ -14,6 +14,7 @@ pub struct SocketForwarder {
     rx: UnboundedReceiver<Message>,
     pub stream: ShardStream,
     tx: UnboundedSender<Message>,
+    closed: bool,
```

Comment:
Is there any chance that this could try and get set by multiple threads at once?